### PR TITLE
Add back navigation support

### LINKED ISSUE #283: Remove back navigation option in Signup and Login screen.
# Why
In the signup and login process we currently catch every "back navigation" event and show a "Abort Signup/Login?" dialog. 

This "back navigation" event is also tracked in the signup and login screen, but here a "Abort Signup/Login?" dialog makes no sense because there is nothing to abort.

# TODOS
- [x] Remove the "back navigation" event tracking for the signup and login screen.
- [ ] Add the new functionality (removing of back navigation tracking) to the dev panel.

# Acceptance criteria

- [ ] A user should be able to return to the page where he is coming from by just navigating back.
(E.g. corbado.com -> app.corbado.com -> "back navigation" -> corbado.com)

### PR DESCRIPTION
**Pull request by Aby-JS**
This pull request adds back navigation support to the application

### CODE REVIEW

File: packages/shared-ui/src/flowHandler/processHandler.ts [hunk 1]
```diff
@@ -226,7 +235,7 @@ export class ProcessHandler {
     return;
   }
 
-  #updatePrimaryBlock = (newPrimaryBlock: Block<unknown>) => {
+  #updatePrimaryBlock = (newPrimaryBlock: Block<unknown>, shouldUpdateUrl = true, forcePush = false) => {
```
Comment (incorbador) [inline packages/shared-ui/src/flowHandler/processHandler.ts]: Why is it called forcePush, isn't this just a normal push to history?
Also I think we should now explain a bit how we set those two values in certain situations:
- when are we setting shouldUpdateUrl to false?
- when are we using replace over push?

**Review by incorbador — COMMENTED** [review]
Added one comment. In addition to that: we now mostly use replace over push (previously we pushed all the time). This means that now it's not possible to navigate back in some situations where it has been possible so far. Examples:
- In signup: from email verify to passkey append
- In signup: from identifierEdit to passkey append (or email verify)

@Adam3371 please have a look at that also.
=> these errors are probably also responsible for (some of) the breaking e2e tests.

File: packages/react/src/contexts/CorbadoSessionContext.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ export interface CorbadoSessionContextProps {
   loading: boolean;
   isAuthenticated: boolean;
   user: SessionUser | undefined;
+  skipHashedUrls: boolean;
```
Comment (incorbador) [inline packages/react/src/contexts/CorbadoSessionContext.tsx]: Why is this part of the react session context? To me this is a setting like any other setting (e.g. projectId or setShortSessionCookie) => it should be in CorbadoAppParams.

Or am I missing something?

Additionally, I would suggest a name like "handleNavigationEvents: boolean" (true ist the default and a user could set this to false to deactivate navigation handling). I think that name is easier to understand and more flexible.

Comment (Aby-JS) [reply packages/react/src/contexts/CorbadoSessionContext.tsx]: you are right.. let me make this change

Comment (incorbador) [reply packages/react/src/contexts/CorbadoSessionContext.tsx]: @Aby-JS I tried to use the setting but it's not available to me in CorbadoAuth and the other components => Did you test this and if yes how can I use the setting?

Comment (Aby-JS) [reply packages/react/src/contexts/CorbadoSessionContext.tsx]: apologies for not testing and missing this. Fixed and tested. You can test now

**Review by incorbador — APPROVED** [review]

**Pull request merged**